src/Symfony/Bundle/FrameworkBundle/composer.json

```diff
@@ -27,7 +27,8 @@
         "symfony/polyfill-mbstring": "~1.0",
         "symfony/filesystem": "~3.4|~4.0",
         "symfony/finder": "~3.4|~4.0",
-        "symfony/routing": "^4.1"
+        "symfony/routing": "^4.1",
+        "symfony/validator": "^4.1"
```

Comment:
-1 for making it a hard requirement. This defeats the work done in 3.4 to reduce the number of mandatory deps in a Symfony project (and defeats the auto-enabling of the component as it would now always be installed)

Reply:
you might want to duplicate the regex in the descriptor instead to avoid a strong dependency (and maybe add a unit test ensuring that the regex stays in sync, but this might be a pain if different versions get installed for FrameworkBundle and the component)

Reply:
What about an interface where you can have an adapter for to validate using the validation component instead? Making the "url is invalid" feature automatically enable when the validation component exists, and do nothing when it isn't (perhaps with a suggestion to install it)

Reply:
well, if you only want to disable the feature, there is no need for an adapter. You would only need to make an existence check in the constructor, and skip the validation if you have `null` as validator.

Reply:
It's nice for other frameworks if they wish to provide an integration without the Symfony validator

Reply:
this descriptor is in FrameworkBundle. That's the package which is not meant to be reused by other frameworks, as it defines the Symfony framework experience.